File: DevEnvironment/Base/src/main/java/base/core/PathBuilder.java
package base.core;

/**
 * The PathBuilder class implements a builder of a path.
 */
public final class PathBuilder extends AbstractPathBuilder {
    private static final String ROOT_DIRECTORY = null;
    private static final String SEPARATOR = WindowsPath.DIRECTORY_SEPARATOR;

    /**
     * The PathBuilder constructor.
     */
    public PathBuilder() {
        this(
            ROOT_DIRECTORY,
            SEPARATOR);
    }

    /**
     * The PathBuilder constructor.
     */
    private PathBuilder(String rootDirectory) {
        super(
            rootDirectory,
            SEPARATOR);
    }

    /**
     * The PathBuilder constructor.
     */
    public PathBuilder(String rootDirectory, String separator) {
        super(
            rootDirectory,
            separator);
    }
}

File: SystemCore/Base/core/PathBuilder.cpp
#include "PathBuilder.h"

#include "WindowsPath.h"

using namespace base;

/**
 * Creates a new path builder.
 */
IPathBuilderSharedPtr Make()
{
    return std::make_shared<PathBuilder>();
}

/**
 * The PathBuilder constructor.
 */
PathBuilder::PathBuilder(const std::wstring& rootDirectory) :
    AbstractPathBuilder(
        rootDirectory,
        environment::windows::WindowsPath::DirectorySeperator())
{
}

/**
 * The PathBuilder destructor.
 */
PathBuilder::~PathBuilder()
{
}
